src/ripple/app/ledger/LedgerToJson.h

```diff
@@ -129,49 +132,84 @@ void fillJson (Object& json, LedgerFill const& fill)
             {
                 if (type == SHAMapTreeNode::tnTRANSACTION_NM)
                 {
-                    SerializerIterator sit (item->peekSerializer ());
-                    STTx txn (sit);
-                    txns.append (txn.getJson (0));
+                     if (bBinary)
+                    {
+                        auto&& obj = appendObject (txns);
+                        obj[jss::tx_blob] = strHex (item->peekData ());
+                    }
+                    else
+                    {
+                        SerializerIterator sit (item->peekSerializer ());
+                        STTx txn (sit);
+                        txns.append (txn.getJson (0));
+                    }
                 }
                 else if (type == SHAMapTreeNode::tnTRANSACTION_MD)
                 {
-                    SerializerIterator sit (item->peekSerializer ());
-                    Serializer sTxn (sit.getVL ());
-
-                    SerializerIterator tsit (sTxn);
-                    STTx txn (tsit);
-
-                    TransactionMetaSet meta (
-                        item->getTag (), ledger.getLedgerSeq(), sit.getVL ());
-                    Json::Value txJson = txn.getJson (0);
-                    txJson[jss::metaData] = meta.getJson (0);
-                    txns.append (txJson);
+                    if (bBinary)
+                    {
+                        SerializerIterator sit (item->peekSerializer ());
+
+                        auto&& obj = appendObject (txns);
+                        obj[jss::tx_blob] = strHex (sit.getVL ());
+                        obj[jss::meta] = strHex (sit.getVL ());
+                    }
```

Comment:
I see that getVL also increments the iterator. That's strange semantics. Accessing an iterator doesn't normally increment it as well. Not related to new code, so don't expect a change.